shell/platform/android/io/flutter/embedding/engine/FlutterJNI.java

```diff
@@ -1258,4 +1258,15 @@ private void ensureRunningOnMainThread() {
   public interface AsyncWaitForVsyncDelegate {
     void asyncWaitForVsync(final long cookie);
   }
+
+  /**
+   * A factory for creating {@code FlutterJNI} instances. Useful for FlutterJNI injections during
+   * tests.
+   */
+  public static class Factory {
```

Comment:
nit: The pattern should have the Factory being an interface.  It's not a huge deal here, but helps if anyone tries to add methods.

Reply:
it creates a bit more indirection to be able to instantiate one if it was an interface though